vtex/pickup-points-modal
# CHK-599: Fix rendering of markers not updating on address change

#### What is the purpose of this pull request?

This pull-request updates the rendering logic for the Map component to update 
the marker icon whenever the user changes their address. This actually is a 
much bigger refactor than I initially planned, but I've done it this way so the 
component is more maintainable across changes.

Basically, I re-implemented how the Map component works. I took much 
inspiration by the [react-google-maps](https://github.com/tomchentw/react-google-maps) library and 
its API. I created the `GoogleMap` and `GoogleMarker` components that were 
necessary for our use case, and moved all the rendering logic for markers into 
the `GoogleMarker` component, so we can render it declaratively instead of 
imperatively using refs.

So now, instead of needing to manually create the `google.maps.Marker` object 
and attach it to the map by calling `marker.setMap(this.map)`, we can simply 
render the `<GoogleMarker />` component as a children of `<GoogleMap />` and it 
will work ootb.

```tsx
<GoogleMap>
  <GoogleMarker location={[lng, lat]} url={markerIcon} />
</GoogleMap>
```

With this, I deleted most of the code inside the `Map` component which was 
related to this imperative rendering, and replaced it with the components, 
which will now be handled by React itself.  I believe it is now a much simpler 
component to understand and to change (going from ~800 LOC down to ~300).

But this change isn't without is compromises, now that I basically 
re-implemented it, there are some flows I might not be aware of that was 
handled with the old code that could not be handled with the new one.

~I still pretend to run the e2e tests with my changes, but haven't done that 
yet.~

Update: I've executed the e2e tests and they all passed, with the exception of two
tests that are always failing.

#### What problem is this solving?

Fixes the updating of marker icons when the address is changed. Before this, 
whenever you updated your address, the best pickup points of the previous 
location would still appear highlighted and the new ones wouldn't.

#### How should this be manually tested?

[Workspace](https://lucas--paguemenos.myvtex.com/checkout/cart/add/?sku=19449&qty=1&seller=1&sc=1)

You can use the postal code `60425-812`.

To see the issue resolved, simply change to a new address in Fortaleza and all 
the markers on the map should be updated accordingly. You can use for example 
the address `av treze de maio 1383`.

#### Screenshots or example usage

#### Types of changes
